CRM/Sepa/Logic/Queue/Close.php

```diff
@@ -56,23 +63,30 @@ public static function launchCloseRunner($txgroup_ids, $target_group_status, $ta
       // count the contributions and create an appropriate amount of items
       $contribution_count = CRM_Core_DAO::singleValueQuery("SELECT COUNT(contribution_id) FROM civicrm_sdd_contribution_txgroup WHERE txgroup_id={$txgroup['id']}");
       $contribution_count += SDD_CLOSE_RUNNER_BATCH_SIZE; // security margin
-      for ($offset=0; $offset <= $contribution_count; $offset+=SDD_CLOSE_RUNNER_BATCH_SIZE) {
+      for ($offset=0; $offset <= $contribution_count; $offset += SDD_CLOSE_RUNNER_BATCH_SIZE) {
         $queue->createItem(new CRM_Sepa_Logic_Queue_Close('update_contribution', $txgroup, $target_contribution_status, $offset));
       }
 
       // finally: render XML and mark the group
-      $queue->createItem(new CRM_Sepa_Logic_Queue_Close('create_xml', $txgroup, $target_group_status));
+      if ($is_received_runner) {
+        $queue->createItem(new CRM_Sepa_Logic_Queue_Close('create_xml', $txgroup, $target_group_status));
```

Comment:
According to the [docs](https://docs.civicrm.org/dev/en/latest/framework/queues/#2-create-itemstasks-on-the-queue), a queue should be filled with `CRM_Queue_Task` objects when it is supposed to be run using CiviCRM's queue runners. Not sure how much BC there is for the concept being used here, though …